### PR TITLE
[BW-764] Use AuthSession for Communication

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### _Summary_

We decided to switch over to AuthSession for many benefits including shared cookies w standalone browser, more reliable deeplinking via custom scheme, etc.

This PR:
- Switches over to use Expo's `openAuthSessionAsync`
- Removes `appDeeplinkUrl` from `AppMetadata`
- `appCustomScheme` is now required in `AppMetadata`
- Removes `handleResponse` as this is now handled by Expo and consumer won't need to handle response URL manually
- Includes breaking change for consumer, will aim for `1.0.0` for next release

### _How did you test your changes?_
| iOS | Android |
| ------------- | ------------- |
| <video src="https://github.com/user-attachments/assets/17ca0f9e-fd13-4f75-8200-05fdf8061b28" /> | <video src="https://github.com/user-attachments/assets/f3e281a4-0b7d-471a-aa33-bb0726a9c5d1" /> |

### CODE REVIEW

File: packages/client/src/components/communication/postRequestToWallet.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @returns A promise that resolves to the response.
+ */
+export async function postRequestToWallet(
+  request: RPCRequestMessage,
+  appCustomScheme: string,
+  wallet: Wallet
+): Promise<RPCResponseMessage> {
+  const { type, scheme } = wallet;
+
+  if (type === 'webBased') {
```
Comment (cb-jake) [inline packages/client/src/components/communication/postRequestToWallet.ts:23]: nit: can this just be `web`?

File: packages/client/src/MWPClient.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ const ACCOUNTS_KEY = 'accounts';
 const ACTIVE_CHAIN_STORAGE_KEY = 'activeChain';
 const AVAILABLE_CHAINS_STORAGE_KEY = 'availableChains';
 const WALLET_CAPABILITIES_STORAGE_KEY = 'walletCapabilities';
-import * as Communicator from './components/communicator';
+import * as Communicator from './components/communication';
```
Comment (cb-jake) [inline packages/client/src/MWPClient.ts:19]: nit, can we just import `postRequestToWallet` directly?

File: packages/client/src/components/communication/index.ts [hunk 3]
```diff
@@ -1,2 +1 @@
-export * from './handleResponse';
```
Comment (cb-jake) [inline packages/client/src/components/communication/index.ts:1]: Can we remove this barrel file. Not great for tree shaking

File: packages/client/src/interfaces/eip1193/EIP1193Provider.test.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ describe('EIP1193Provider', () => {
     (MWPClient.createInstance as jest.Mock).mockResolvedValue(mockClient);
 
     provider = new EIP1193Provider({
-      metadata: { appName: 'Test App', appDeeplinkUrl: 'test://deeplink' },
+      metadata: { appName: 'Test App', appCustomScheme: 'test://deeplink' },
```
Comment (cb-jake) [inline packages/client/src/interfaces/eip1193/EIP1193Provider.test.ts:33]: If we're publishing a new version do we really need the `app` prefix on these vars?

can we just same `name`, `scheme`, etc...? feels a tad verbose

File: packages/client/src/MWPClient.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ const ACCOUNTS_KEY = 'accounts';
 const ACTIVE_CHAIN_STORAGE_KEY = 'activeChain';
 const AVAILABLE_CHAINS_STORAGE_KEY = 'availableChains';
 const WALLET_CAPABILITIES_STORAGE_KEY = 'walletCapabilities';
-import * as Communicator from './components/communicator';
+import * as Communicator from './components/communication/postRequestToWallet';
```
Comment (cb-jake) [inline packages/client/src/MWPClient.ts:19]:
```suggestion
import { postRequestToWallet } from './components/communication/postRequestToWallet';
```

File: packages/client/src/core/message/RPCMessage.ts [hunk 6]
```diff
@@ -40,5 +39,8 @@ export interface RPCResponseMessage extends RPCMessage {
 
 type RequestAccountsAction = {
   method: 'eth_requestAccounts';
-  params: Pick<AppMetadata, 'appName' | 'appLogoUrl'>;
+  params: {
+    appName: string;
+    appLogoUrl?: string;
```
[the remark this answers is not in the document]
Comment (fan-zhang-sv) [reply packages/client/src/core/message/RPCMessage.ts:44]: yea, scw client is expecting app prefix from both web sdk and mobile sdk, so keeping it like this for simplicity

**Review by cb-jake — APPROVED** [review]

**Pull request merged**